library/X509/CertificateUtils.php

```diff
@@ -151,16 +152,31 @@ private static function shortNameFromDN(array $dn)
     /**
      * Split the given Subject Alternative Names into key-value pairs
      *
-     * @param   string  $sans
+     * @param string $sans
      *
-     * @return  \Generator
+     * @return  Generator
      */
-    private static function splitSANs($sans)
+    private static function splitSANs(string $sans): Generator
```

Comment:
To be honest, I don't remember why this was a generator in the first place, but I don't think it's useful to use one here 😆.

Anyway, I don't like this code. What about something like this (note that I tested this quickly on the CLI so I didn't use `Str::trimSplit()`):

```
$sanStr = 'IP Address:192.168.15.1, Registered ID:1.1.1.1, othername:<unsupported>, email:rfc@rfc.de, DNS:dom.ain, DNS:oth.er';
$san = [];
foreach (explode(',', $sanStr) as $altName) {
    list($k, $v) = explode(':', $altName);
    $san[trim($k)][] = trim($v);
}
$order = array_flip(['DNS', 'URI', 'IP Address', 'email']);
uksort($san, function ($a, $b) use ($order) {
    return ($order[$a] ?? PHP_INT_MAX) <=> ($order[$b] ?? PHP_INT_MAX);
});
```

Note that I also adjusted the order. `rfc822Name` is `mail`. Which in my opinion is less useful than `IP` and `URI`.